### DISCUSSION TITLE
Changed Tesla Password - Stopped Collecting Data

### DISCUSSION THREAD

**Discussion by mbaturin** [jasonacox/Powerwall-Dashboard#509] [General]
I changed my Tesla password a week ago and didn't think to update the Powerwall Dashboard config.  Is still worked for the last week, but suddenly today stopped collecting data from around 5:35PM ET.  I reran the setup and updated the password, but I am still not having any luck getting it to connect.  My logs indicate that there are timeout errors:

[LogsPowerwallDashboard.txt](https://github.com/user-attachments/files/16742590/LogsPowerwallDashboard.txt)

One thing I noticed is that if I access my Gateway from a browser (pr at least the page now saying to download Tesla One), I see that it is not https.  Log file is saying its   using port=443 which makes me think it is trying over https?

<img width="277" alt="image" src="https://github.com/user-attachments/assets/17d5b3a8-5603-4f21-b9c5-0f49c7c9ddd9">

verify.sh found these failures:

Verify Powerwall-Dashboard 4.4.6 on Linux - Timezone: America/New_York
----------------------------------------------------------------------------
This script will attempt to verify all the services needed to run
Powerwall-Dashboard. Use this output when you open an issue for help:
https://github.com/jasonacox/Powerwall-Dashboard/issues/new

Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): ERROR: Not Listening
 - Version: Unknown
 - Powerwall State: ERROR: Not Connected - Firmware: Unknown
 - Powerwall Gateway TEDAPI: Not Available (192.168.91.1)
 - Cloud Mode: NO

Checking telegraf
----------------------------------------------------------------------------
 - Config File telegraf.conf: GOOD
 - Local Config File telegraf.local: GOOD
 - Container (telegraf): GOOD
 - Version: Telegraf 1.28.2 (git: HEAD@8d9cf395)

Checking influxdb
----------------------------------------------------------------------------
 - Config File influxdb.conf: GOOD
 - Environment File influxdb.env: GOOD
 - Container (influxdb): GOOD
 - Service (port 8086): GOOD
 - Filesystem (./influxdb): GOOD
 - Version: InfluxDB shell version: 1.8.10

Checking grafana
----------------------------------------------------------------------------
 - Config File grafana.env: GOOD
 - Container (grafana): GOOD
 - Service (port 9000): GOOD
 - Filesystem (./grafana): GOOD
 - Version: Grafana CLI version 9.1.2

Checking weather411
----------------------------------------------------------------------------
 - Container (weather411): GOOD
 - Service (port 8676): GOOD
 - Weather: {"temperature": 74.39}
 - Version: 0.2.3

One or more tests failed.

**Comment by jasonacox** [comment]
Hi @mbaturin , sorry to hear about the trouble. It is also possible that your system upgraded to a newer firmware that eventually caused the lockout.

> I see that it is not https

Actually, your screenshot shows "https:" but it is red because it is a Tesla self-signed cert. This is as expected.

> I reran the setup and updated the password

Keep in mind that the password for the Powerwall is not the same as your Tesla online account. You should try to log in to the web portal of your Powerall. You will need to bypass the the "upgrade" screens by (1) click the "Continue" button at the bottom then (2) click the "UPGRADE LATER" button (super light color button on bottom of second screen). It should then present you with a login. The login you use for that will be the same you use for pypowerwall setup.  Note: If you are getting LoginError: Invalid Powerwall Login errors but have double checked your password and are sure it is correct, try using the last 5 characters of the password written on the Powerwall Gateway.

> [LogsPowerwallDashboard.txt](https://github.com/user-attachments/files/16742590/LogsPowerwallDashboard.txt)

The logs do indicate the Powerwall is not responding to you. That could be caused by being locked out (the Powerwall will block you if it detects heavy load).  After you verify the password above, stop pypowerwall and let it stay offline for at least 10 minutes. Restart it and check the logs to ensure it is working:

```bash
docker stop pypowerwall
# wait 10m
docker start pypowerwall
# check logs
docker logs pypowerwall -f
```

If it is still not working, try installing pypowerwall and running a scan to ensure it can find the powerwall:

```bash
pip install pypowerwall
python3 -m pypowerwall scan
```

**Reply by mbaturin** [reply]
Thanks for the reply.  I did what you suggested and went to the gateway IP to login (never noticed the Continue button at the bottom before).  Here is where it gets very strange.  You were right that the login was the last 5 of the password on the Gateway, but last night I got the data to start flowing again once I redid setup with that password, but then again with my Tesla account username and password.  After I did setup with the last 5, I saw the log said invalid credentials.  So I changed it back to the Tesla accounts new username and password (it was the old Tesla Account username and password for the last month) and it started working again.  I checked the pyenvironment.sh file and it currently has the Tesla account username and password and data is flowing in.  However, I can only login to the gateway directly using the same email but last 5 of PW written on the gateway as the password.

**Reply by jasonacox** [reply]
Fascinating! I wonder if they added a feature to allow the Tesla credentials to work for the local login. The odd thing is that pypowerwall in `local` mode is basically using the same API as the web portal. I do wonder if you setup is using `cloud` mode instead.

Can you run the `verify.sh` command again?  Or provide the first lines of the log after a pypowerwall restart?

**Reply by mbaturin** [reply]
Here is the verify.sh output as of now:

Verify Powerwall-Dashboard 4.4.6 on Linux - Timezone: America/New_York
----------------------------------------------------------------------------
This script will attempt to verify all the services needed to run
Powerwall-Dashboard. Use this output when you open an issue for help:
https://github.com/jasonacox/Powerwall-Dashboard/issues/new


Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): GOOD
 - Version: 0.10.9 Proxy t63
 - Powerwall State: CONNECTED - Firmware: 24.20.0
 - Site Name: Batcave Arc Reactor
 - Powerwall Gateway TEDAPI: Not Available (192.168.91.1)
 - Cloud Mode: NO

Checking telegraf
----------------------------------------------------------------------------
 - Config File telegraf.conf: GOOD
 - Local Config File telegraf.local: GOOD
 - Container (telegraf): GOOD
 - Version: Telegraf 1.28.2 (git: HEAD@8d9cf395)

Checking influxdb
----------------------------------------------------------------------------
 - Config File influxdb.conf: GOOD
 - Environment File influxdb.env: GOOD
 - Container (influxdb): GOOD
 - Service (port 8086): GOOD
 - Filesystem (./influxdb): GOOD
 - Version: InfluxDB shell version: 1.8.10

Checking grafana
----------------------------------------------------------------------------
 - Config File grafana.env: GOOD
 - Container (grafana): GOOD
 - Service (port 9000): GOOD
 - Filesystem (./grafana): GOOD
 - Version: Grafana CLI version 9.1.2

Checking weather411
----------------------------------------------------------------------------
 - Container (weather411): GOOD
 - Service (port 8676): GOOD
 - Weather: {"temperature": 88.34}
 - Version: 0.2.3

All tests succeeded.

Also confirmed that the pypowerwall.env file has the email and pw of my Tesla account, not the last 5 of the gateways password on the sticker.  

Data has been flowing in fine!

<img width="694" alt="image" src="https://github.com/user-attachments/assets/2bf1e64a-c907-4414-b075-6544f711b6b0">

**Reply by jasonacox** [reply]
That's fantastic news, @mbaturin !  Nice look graphs... my 8.5kW system looks pretty pathetic next to your generation numbers. 😜  

I should probably add a note to the troubleshooting tips to try the Tesla Account now.

**Reply by mbaturin** [reply]
> That's fantastic news, @mbaturin ! Nice look graphs... my 8.5kW system looks pretty pathetic next to your generation numbers. 😜
> 
> I should probably add a note to the troubleshooting tips to try the Tesla Account now.

Thanks!  My system is 23.49 PV and 20.9 Inverter.  It's strange I still cannot connect via the gateway IP with anything but the last 5 of the gateway password, but Powerwall-Dashboard works with the Tesla creds.  Thanks for building this awesome app!

**Comment by mbaturin** [comment]
@jasonacox  - today my power went out for less than a second and came back on.  I have a ticket in with Tesla Powerwall support to put a -2.5 offset on my frequency to fix issues with home UPS's I have.  I thought it had to do with that (maybe it did).  However, when this happened my Powerwall Dashboard stopped collecting data.  I can't login to the Gateway from the local IP, nor from Netzero app etc.  I ran a verify on powerwall-dashboard and see this now:

Verify Powerwall-Dashboard 4.4.6 on Linux - Timezone: America/New_York
----------------------------------------------------------------------------
This script will attempt to verify all the services needed to run
Powerwall-Dashboard. Use this output when you open an issue for help:
https://github.com/jasonacox/Powerwall-Dashboard/issues/new


Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): ERROR: Not Listening
 - Version: Unknown
 - Powerwall State: ERROR: Not Connected - Firmware: Unknown
 - Powerwall Gateway TEDAPI: Not Available (192.168.91.1)
 - Cloud Mode: NO

Checking telegraf
----------------------------------------------------------------------------
 - Config File telegraf.conf: GOOD
 - Local Config File telegraf.local: GOOD
 - Container (telegraf): GOOD
 - Version: Telegraf 1.28.2 (git: HEAD@8d9cf395)

Checking influxdb
----------------------------------------------------------------------------
 - Config File influxdb.conf: GOOD
 - Environment File influxdb.env: GOOD
 - Container (influxdb): GOOD
 - Service (port 8086): GOOD
 - Filesystem (./influxdb): GOOD
 - Version: InfluxDB shell version: 1.8.10

Checking grafana
----------------------------------------------------------------------------
 - Config File grafana.env: GOOD
 - Container (grafana): GOOD
 - Service (port 9000): GOOD
 - Filesystem (./grafana): GOOD
 - Version: Grafana CLI version 9.1.2

Checking weather411
----------------------------------------------------------------------------
 - Container (weather411): GOOD
 - Service (port 8676): GOOD
 - Weather: {"temperature": 77.68}
 - Version: 0.2.3

One or more tests failed.

Any ideas as to how I might be able to get this back on?

EDIT: Spoke with Powerwall support - got info that my system was updated this morning to 24.28.1.  That likely explains the very quick power fluctuation, but hoping they didn't remove API access in this version.

EDIT2: After a few hours I was finally able to get the password to update on the Gateway and then was able to use that to rerun setup.sh and put the new password in.  Everything worked after that.  I feel like I did the same thing 5-6 times and it worked on the 6th time, so I think it might just be that I had to wait some time.  As a sidenote, confirmed working on 24.28.1.

**Reply by jasonacox** [reply]
Thanks @mbaturin - One thought is that I have seen the Powerwall rate limit or block pypowerwall after several attempts to log in with the old password. It requires up to 10 minutes for this to clear. Since you had a running pypowerwall, it would have attempted to retry login every 5s or so which could have caused this condition. There is logic in pypowerwall to do a cool-down mode if it is getting 4xx errors from the login path, but it could be that during the upgrade the Powerwall was just being unresponsive versus giving error responses. You might be able to see details doing `docker logs pypowerwall` if you were curious. :) 

Glad it is working again for you!
